Add tests for components

bini-i opened the pull request:
### The following tests were added in this branch
- add snapshot tests for App component
- add snapshot tests for Home component
- add snapshot tests for Quote component
- add unit tests for calculate.js functionalities

bini-i committed:
Add tests for components
- add snapshot tests for App component
- add snapshot tests for Home component
- add snapshot tests for Quote component
- add unit tests for calculate.js functionalities

rindrajosia commented on src/logic/operate.js:
```diff
@@ -12,7 +12,6 @@ function operate(numberOne, numberTwo, operation) {
-    window.result = result;
   }
   if (operation === "÷") {
     if (numberTwo === 0) return "Error: Division by zero";
```
- [ ] [REQUIRED] Please add a test unit for this operation as I got an error related to proptype when I divide number by 0

rindrajosia reviewed (changes requested):
Hi, 

Good job so far!
There are some issues to revise before continuing to the next project but hang in there and read the comments below with care. You can do it! :muscle:


## Required Changes ♻️

- [ ] Please update your live demo and add information about `how to use ` your project in your readme to make it more descriptive.
- [ ] all functionalities should be tested using [React Testing Library](https://testing-library.com/docs/react-testing-library/intro/), @testing-library/react, see the [screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/react_reducx/testing/useReactTestingLibrary.png) from [Pathwright](https://microverse.pathwright.com/library/react-redux-v11/184438/path/step/106580952/).
- [ ] Please make sure that all snapshots are up to date and none is obsolete.
![1](https://user-images.githubusercontent.com/58552748/129872752-b4837088-23f5-476f-9922-a28310740ce8.png)

- [x] I got an error related to the propTypes when I divide a number by 0 (zero). Keep in mind that PropTypes are a mechanism to ensure that components use the correct data type and pass the right data, and that components use the right type of props, and that receiving components receive the right type of props.
![1](https://user-images.githubusercontent.com/58552748/129873578-247218ea-7199-4714-b130-eb75a9d95f2c.gif)

- [ ] It will be nice if users can make an operation with the previous result.
![2](https://user-images.githubusercontent.com/58552748/129873845-0e6e9c52-cb7e-4ea4-99ac-e35fa8a2f04d.gif)


_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

bini-i committed:
update App.js and Dispay.js
- update next proptype to a string in Display.js
- pass next prop to Display component as string in App.js

bini-i committed:
Update REAMDE.md
- add how to use and how run tests section

bini-i committed:
Remove obsolete tests
- remove obsolete tests from src/components/__test__

bini-i committed:
update App.test.js, Home.test.js and Quote.test.js
- add render from react-testing-library in all test files and test rendering of all components

george-shammar reviewed (approved):
Hi @bini-i , 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
